src/components/forms/inputs/TextInput.jsx

```diff
@@ -18,7 +14,7 @@ export default class TextInput extends InputBase {
                 { id: this.props.placeholder });
         }
 
-        const maxLength = this.props.maxLength ? this.props.maxLength : 0
+        const maxLength = this.props.maxLength ? this.props.maxLength : null
```

Comment:
Agree with Olle here, this is superfluous because if this.props.maxLength is not supplied as a prop, this.props.maxLength will be undefined, and in Javascript null == undefined (though nul !== undefined)

Reply:
Again I must ask, are you sure that React will treat `null` and `undefined` the same way for the `maxLength` property? If we are certain it will, I agree that passing through `this.props.maxLength` is nicer.

Reply:
Just so I understand this correctly. Should I remove the variable with ternary and change `maxLength` to`this.props.maxLength` in return?

Reply:
Yeah, exactly! Verify that it works, and if it does, that's the cleanest solution.

Reply:
Done!

Reply:
Yup, that's correct!

Reply:
> Again I must ask, are you sure that React will treat `null` and `undefined` the same way for the `maxLength` property? If we are certain it will, I agree that passing through `this.props.maxLength` is nicer.

I tried it out in a react sandbox and it worked fine.